cishouseholds/validate.py

```diff
@@ -148,3 +152,70 @@ def check_singular_match(
             failure_column_name, F.when(F.col(existing_failure_column) == 1, 1).otherwise(F.col(failure_column_name))
         )
     return df.drop("TOTAL")
+
+
+class ConfigError(Exception):
+    pass
+
+
+def upfront_key_value_parameters_validation(all_function_dict: Dict, config: dict):
+    """
+    Checks that there's a valid input in the pipeline_config.yaml for every stage
+    input argument.
+
+    Parameters
+    ----------
+    all_function_dict: dictionary of all functions name and function object in pipeline_stages.py
```

Comment:
The type is usually after the colon in [this docstring format](https://numpydoc.readthedocs.io/en/latest/format.html), while the description goes on a new line indented